### PR TITLE
docs: Core api reference

### PR DESCRIPTION
**Pull request by lukaszjedrasik**
### Related Issues

closes # [EN-198](https://vsf.atlassian.net/browse/EN-198?atlOrigin=eyJpIjoiYTRlZTA1YzNlM2YzNDYxN2FmYzJmMTliYzMzNTE0OGQiLCJwIjoiaiJ9)

### Short Description of the PR

- added reference generator based on `api-extractor` config files 
- added generated documentation for core API reference
- added a few examples of how to add comments generate into `Description` ([TSDOC](https://tsdoc.org/))

### Screenshots of Visual Changes before/after (if There Are Any)

https://user-images.githubusercontent.com/40273258/117977274-eebb8180-b330-11eb-80c2-bc7ce85ff63c.mp4

### CODE REVIEW

**Comment by filrak** [comment]
Is it included into build process?

**Comment by lukaszjedrasik** [comment]
> Is it included into build process?

Yes, it is added to our `deploy-docs` workflow like API reference for CT

File: packages/core/core/src/types.ts [hunk 1]
```diff
@@ -632,6 +638,9 @@ export interface FactoryParams {
   provide?: (context: Context) => any;
 }
 
+/**
+ * @internal
+ */
```
Comment (filipsobol) [inline packages/core/core/src/types.ts:643]:
```suggestion
```

File: packages/core/core/src/types.ts [hunk 2]
```diff
@@ -643,6 +652,9 @@ export interface CallHookParams<C> extends HookParams<C> {
 export type BeforeCallArgs = any;
 export type AfterCallArgs = any;
 
+/**
+ * @internal
+ */
```
Comment (filipsobol) [inline packages/core/core/src/types.ts:657]:
```suggestion
```

**Review by filipsobol — APPROVED** [review]

**Pull request merged**